org.eclipse.ice.commands/src/main/java/org/eclipse/ice/commands/ConnectionManager.java

```diff
@@ -35,45 +52,235 @@ public ConnectionManager() {
 	 */
-	public boolean closeConnection(Connection connection) {
+	public void removeConnection(String connectionName) {
+		// Remove it from the list of connections
+		connectionList.remove(connectionName);
+	}
 
-		return false;
+	/**
+	 * This function removes all particular connections from the connection list
+	 */
+	public void removeAllConnections() {
+		// Remove all of the items from the hashmap
+		connectionList.clear();
 	}
 
 	/**
 	 * Closes all connections that remain open.
+	 */
+	public void closeAllConnections() {
+		// Iterate over all available connections in the list and disconnect
+		for (Connection connection : connectionList.values()) {
+			connection.getSession().disconnect();
+		}
+
+	}
+
+	/**
+	 * This function lists all the connections (and their statuses, i.e. if open or
+	 * not) to the logger, if so desired. Useful for checking the connections and
+	 * their statuses.
+	 */
+	public void listAllConnections() {
+		// Iterate over all available connections
+		for (String name : connectionList.keySet()) {
+			// Build a message
+			String msg = null;
+			// Get the host for the connection
+			String host = connectionList.get(name).getConfiguration().getHostname();
+			// Get the username for the connection
+			String username = connectionList.get(name).getConfiguration().getUsername();
+			// Check the status. If it is open or closed (i.e. connected or disconnected)
+			String status = "";
+			if (isConnectionOpen(name))
+				status = "open";
+			else
+				status = "closed";
+
+			// Build a message to send to the logger
+			msg = "Connection " + name + ": " + username + "@" + host + " is " + status;
+
+			logger.info(msg);
+		}
+
+	}
+
+	/**
+	 * This function checks whether or not a particular connection is currently
+	 * connected.
+	 * 
+	 * @param connectionName - name of connection to check
+	 * @return - boolean indicating whether or not it is connected (true) or not
+	 *         (false)
+	 */
+	public boolean isConnectionOpen(String connectionName) {
+		Connection connection = getConnection(connectionName);
+		return connection.getSession().isConnected();
+	}
+
+	/**
+	 * This function gets the password from the user as a prompt. It uses the
+	 * {@link org.eclipse.ice.commands.ConsoleEraser#run} method to "erase" the
+	 * characters at the console prompt as they are typed in, so that the password
+	 * isn't shown. The prompt is terminated by a carriage return.
+	 * 
+	 * @return - char array of password chars
+	 */
+	private final char[] getPassword() {
+
+		String password = "";
+		ConsoleEraser eraser = new ConsoleEraser();
+
+		logger.info("Please enter your password: ");
+
+		BufferedReader in = new BufferedReader(new InputStreamReader(System.in));
+
+		eraser.start();
+		try {
+			password = in.readLine();
+			in.close();
+		} catch (IOException e) {
+			logger.error("Couldn't read the password...");
+			return null;
+		}
+
+		eraser.stopErasing();
+		System.out.print("\b");
```

Comment:
Is this required?

Reply:
Yes, it stops the thread from constantly deleting the previous character when someone is inputting their password (so their password doesn't just get typed at the shell, for example). I'll add a comment.